File: src/main/java/br/com/repository/Usuario/FornecedorRepository.java
package br.com.repository.usuario;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import br.com.model.entities.classes.usuario.Fornecedor;

@Repository
public interface FornecedorRepository extends CrudRepository<Fornecedor, Integer> {
    Fornecedor findByCnpjCpf(String cnpjCpf);

    Fornecedor findByEmail(String email);
}

File: src/main/java/br/com/repository/usuario/FornecedorRepository.kt
package br.com.repository.usuario

import br.com.model.usuario.Fornecedor
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FornecedorRepository : JpaRepository<Fornecedor, Int> {
    fun findByCnpjCpf(cnpjCpf: String): Fornecedor
    fun findByEmail(email: String): Fornecedor
}
